cellprofiler/modules/measureradialentropy.py

```diff
@@ -0,0 +1,205 @@
+<hr>
+<p>MeasureRadialEntropy divides an object into pie-shaped wedges and
+ measures either the mean or median intensity of each.  Once the intensity
+ of each wedge has been calculated, the entropy of the bin totals is calculated.</p>
+
+ <p>This module is under construction</p>
+
+'''
+
+
+import numpy as np
+from scipy import stats
+
+
+
+import cellprofiler.module as cpm
+import cellprofiler.measurement as cpmeas
+import cellprofiler.object as cpo
+import cellprofiler.setting as cps
+
+from centrosome.cpmorphology import minimum_enclosing_circle
+
+
+
+ENTROPY = "Entropy"
+
+
+
+class MeasurementTemplate(cpm.Module):
+
+    module_name = "MeasureRadialEntropy"
+    category = "Measurement"
+    variable_revision_number = 1
+
+
+    def create_settings(self):
+
+        self.input_object_name = cps.ObjectNameSubscriber(
+            "Select objects to measure",
+            doc="""Select the objects whose radial entropy you want to measure.""" % globals())
+
+        self.input_image_name = cps.ImageNameSubscriber(
+            "Select an image to measure", doc="""Select the
+            grayscale image you want to measure the entropy of.""" % globals())
+
+        self.bin_number=cps.Integer(
+            "Input number of bins", 6, minval=3, maxval=60,
+            doc="""Number of radial bins to divide your object into.  The minimum number
+            of bins allowed is 3, the maximum number is 60.""" % globals())
+
+        self.intensity_measurement=cps.Choice(
+            "Which intensity measurement should be used?", ['Mean','Median'], value='Mean',doc="""
+            Whether each wedge's mean or median intensity should be used to calculate the entropy.""" % globals())
+
+    def settings(self):
+        return [self.input_image_name, self.input_object_name,
+                self.intensity_measurement, self.bin_number]
+
+
+    def run(self, workspace):
+        #Import the workspace and the current measurements
+        meas = workspace.measurements
+        assert isinstance(meas, cpmeas.Measurements)
+
+        statistics = [["Entropy"]]
+
+        workspace.display_data.statistics = statistics
+
+        #Import the settings
+        input_image_name = self.input_image_name.value
+        input_object_name = self.input_object_name.value
+        metric = self.intensity_measurement.value
+        bins = self.bin_number.value
+
+        image_set = workspace.image_set
+
+        input_image = image_set.get_image(input_image_name,
+                                          must_be_grayscale=True)
+        #Read out the pixel data
+        pixels = input_image.pixel_data
+
+        object_set = workspace.object_set
+        assert isinstance(object_set, cpo.ObjectSet)
+
+        objects = object_set.get_objects(input_object_name)
+        labels = objects.segmented
+
+        indexes = objects.indices
+        #Calculate the center of the objects- I'm guessing there's a better way to do this but this was already here
```

Comment:
Yeah, `centroid` from `scipy` is probably better since we are moving away from `centrosome`.